exchanges/persisted-fetch/src/persistedFetchExchange.ts

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import {
+  Source,
+  fromValue,
+  filter,
+  merge,
+  mergeMap,
+  pipe,
+  share,
+  takeUntil,
+} from 'wonka';
+
+import {
+  CombinedError,
+  Exchange,
+  Operation,
+  OperationResult,
+} from '@urql/core';
+
+import {
+  makeFetchBody,
+  makeFetchURL,
+  makeFetchOptions,
+  makeFetchSource,
+} from '@urql/core/internal';
+
+import { hash } from './sha256';
+
+export const persistedFetchExchange: Exchange = ({ forward }) => {
+  let supportsPersistedQueries = true;
+
+  return ops$ => {
+    const sharedOps$ = share(ops$);
+    const fetchResults$ = pipe(
+      sharedOps$,
+      filter(operation => operation.operationName === 'query'),
+      mergeMap(operation => {
+        const { key } = operation;
+        const teardown$ = pipe(
+          sharedOps$,
+          filter(op => op.operationName === 'teardown' && op.key === key)
+        );
+
+        if (!supportsPersistedQueries) {
```

Comment:
Wondering if this can be true for one query but not another (for example due to microservices)

Reply:
I'm just mirroring what the spec / Apollo Persisted Queries says. But generally, a merged schema should be responsible for the extension, since queries can overlap several micro-services. So by spec, this shouldn't happen. And if it does, this doesn't explode 😅